numpy/_core/src/umath/stringdtype_ufuncs.cpp

```diff
@@ -482,20 +427,24 @@ minimum_strided_loop(PyArrayMethod_Context *context, char *const data[],
         const npy_packed_static_string *sin1 = (npy_packed_static_string *)in1;
         const npy_packed_static_string *sin2 = (npy_packed_static_string *)in2;
         npy_packed_static_string *sout = (npy_packed_static_string *)out;
-        if (_compare(in1, in2, in1_descr, in2_descr) < 0) {
+        int cmp = _compare(in1, in2, in1_descr, in2_descr);
+        if (cmp == 0 && (in1 == out || in2 == out)) {
```

Comment:
@ngoldbaum - this I think is actually a small bug in the current implementation: if the strings compare equal, then no copy should happen if either side is in-place.

Reply:
thanks for catching this!!